ruby/ql/lib/codeql/ruby/frameworks/ActiveStorage.qll

```diff
@@ -17,43 +18,13 @@ class ActiveStorageFilenameSanitizedCall extends Path::PathSanitization::Range,
   }
 }
 
-/** The taint summary for `ActiveStorage::Filename.new`. */
-class ActiveStorageFilenameNewSummary extends SummarizedCallable {
-  ActiveStorageFilenameNewSummary() { this = "ActiveStorage::Filename.new" }
-
-  override MethodCall getACall() {
-    result =
-      API::getTopLevelMember("ActiveStorage")
-          .getMember("Filename")
-          .getAnInstantiation()
-          .asExpr()
-          .getExpr()
-  }
-
-  override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
-    input = "Argument[0]" and
-    output = "ReturnValue" and
-    preservesValue = false
-  }
-}
-
-/** The taint summary for `ActiveStorage::Filename#sanitized`. */
-class ActiveStorageFilenameSanitizedSummary extends SummarizedCallable {
-  ActiveStorageFilenameSanitizedSummary() { this = "ActiveStorage::Filename#sanitized" }
-
-  override MethodCall getACall() {
-    result =
-      API::getTopLevelMember("ActiveStorage")
-          .getMember("Filename")
-          .getInstance()
-          .getAMethodCall("sanitized")
-          .asExpr()
-          .getExpr()
-  }
-
-  override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
-    input = "Argument[-1]" and
-    output = "ReturnValue" and
-    preservesValue = false
+/** Taint related to `ActiveStorage::Filename`. */
+private class Summaries extends ModelInput::SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "activestorage;;Member[ActiveStorage].Member[Filename].Method[new];Argument[0];ReturnValue;taint",
+        "activestorage;;Member[ActiveStorage].Member[Filename].Instance.Method[sanitized];Argument[-1];ReturnValue;taint",
```

Comment:
(not your PR, but `Argument[-1]` does not work; should be `Receiver` instead (actually, it should be `Argument[Receiver]`, but it is the latter for now.)

Reply:
I'd like to update it in this PR so these models can serve as a reference how to do things.

In the process of doing this I came to few realizations:
- Ruby API graphs don't have a label for `Receiver`. I initially wanted to sync these which is easy enough, but due to how the receiver argument works in Ruby, I don't think this will ever be needed in the identifying access path or in API graphs even. So I deliberately left that out. (In JS one can pass an arbitrary value as `this` and some libraries take advantage of that, so the situation is different there).
- `BlockArgument` was not implemented in the identifying access path. Added that in a separate commit.
- There is no support for keyword arguments in neither the identifying access path nor in input/output specs. It's straightforward to add, but we'll need to agree on whether to name it `Argument[name]` or `KeywordArgument[name]`. In Python there is a desire to use `Argument[name]` because almost any positional argument has a corresponding named argument, which we'll want to express with a single token. For now I've left it out.